vpr/src/power/power.cpp

```diff
@@ -1361,14 +1363,13 @@ bool power_uninit() {
     bool error = false;
 
     for (size_t rr_node_idx = 0; rr_node_idx < device_ctx.rr_nodes.size(); rr_node_idx++) {
```

Comment:
Probably something for later, but if we could
```
for (RRNodeId id : device_ctx.rr_nodes) {
}
```

It would be much more expressive what we mean, and also we wouldn't need the typecast that we have to do now when passing it to `rr_graph.node_fan_in(....)`.

(this is somewhat the idea of working towards a goal that makes the code look like one wishes it looked like, so then towards `device_ctx.rr_nodes` being something that iterates out`RRNodeId`.

Reply:
Agreed.  I think that change would work, and would be more expressive (and avoiding casts is good).